### DISCUSSION TITLE
缓存不起作用了

### DISCUSSION THREAD

**Discussion by Domain** [xmake-io/xmake#4382] [Q&A] [answered]
用最新的版本，发现编译缓存不起作用，每次编译都是重新开始，我把所以xmake生成的目录全部删除，还是一样，用xmake f --ccache=y，也没用，不知道还有办法知道哪里出问题了

**Comment by waruqi** [comment, marked as answer]
> 每次编译都是重新开始

这是增量编译问题，跟 build cache 没任何关系

```console
xmake f --policies=diagnosis.check_build_deps
xmake
```

开启增量编译检测，看下 logs ，自己排查下哪个文件一直在 changed

**Reply by Domain** [reply]
不是的，启动了缓存编译不是显示ccache开头的吗？我这里全部文件都没有，我清空生成的文件，重新编译，全部是compiling开头，没有一个文件显示ccache

**Reply by waruqi** [reply]
什么系统？ win?

**Reply by Domain** [reply]
是的，win11

**Reply by Domain** [reply]
刚刚在linux试了一下，缓存正常的

**Reply by waruqi** [reply]
win上默认禁用了 从某个版本开始，因为 win 的 预处理器太慢，并且 win 开独立预处理进程也比较重，开了 build cache 效果不理想，很多情况下 比不开还慢好几倍。。

**Reply by Domain** [reply]
哦，好吧，这就说得通了
